PTGLdynamics/codes/plotting-scripts/calculateChanges.py

```diff
@@ -0,0 +1,386 @@
+             
+            else:
+                log("You entered non valid edge names.", 'i')
+
+    log("Changes for each edge: " + str(changes_edges), 'i')
+
+    # Calculate change for each chain
+    for key in changes_edges:
+        nodes = key.split(' ')
+        node_1 = nodes[0].replace(' ', '').replace('{', '')
+        node_2 = nodes[1].replace(' ', '').replace('}', '')
+
+    
+        # Exclude chains from calculation
+        if (node_1 in exclude_chains) or (node_2 in exclude_chains):
+            pass
+        
+        else:
+            old_value_node_1 = changes.get(node_1)
+            old_value_node_2 = changes.get(node_2)
+            if old_value_node_1 == None:
+                old_value_node_1 = 0
+            if old_value_node_2 == None:
+                old_value_node_2 = 0
+            new_value_node_1 = int(old_value_node_1) + changes_edges[key]
+            new_value_node_2 = int(old_value_node_2) + changes_edges[key]
+    
+            changes[node_1] = new_value_node_1
+            changes[node_2] = new_value_node_2
+            
+    # Divide by chainlength
+    if(divide_chainlength != ''):
+        changes_divided = divide_by_chainlength(changes, divide_chainlength)
+        changes = changes_divided        
+
+    log("Changes for each chain: " + str(changes), 'i')
+  
+# Create output csv file.
+changes_out = open(output_dir + '/' + 'changes_each_' + calculate + '_based_on_' + based_on + '_frame' + str(args.first_timestep) + '_to_frame' + str(args.last_timestep) + '.csv','w')
+
+if calculation == "(chain, res)":
+    changes_out.write("chain" + "," +  "change in res-res contacts" + '\n')
+elif calculation == "(res, res)":
+    changes_out.write("residue (PDB_ID|chain|iCode)" + "," +  "change in res-res contacts" + '\n')
+elif calculation == "(chain, CG)":
+    changes_out.write("chain" + "," +  "change in CG´s" + '\n')
+
+for key in changes:
+    changes_out.write(key + ',' + str(changes[key]) + '\n')
+
+changes_out.close()
+
+all_changes = set(changes.values())
+value_max = int(max(all_changes))
+value_min = int(min(all_changes))
+change = value_max - value_min
+
+for key in changes:
+    value = changes.get(key)
+    percent = (int(value) - value_min) / change
+    percent = round(percent, 2)
+    changes[key] = percent
+    
+percents = open(output_dir + '/' + 'changes_in_percent_each_' + calculate + '_based_on_' + based_on + '_frame' + str(args.first_timestep) + '_to_frame' + str(args.last_timestep) + '.csv','w')
+
+if calculation == "(chain, res)":
+    percents.write("chain" + "," +  "change in res-res contacts in percent" + '\n')
+elif calculation == "(res, res)":
+    percents.write("residue (PDB_ID|chain|iCode)" + "," +  "change in res-res contacts in percent" + '\n')
+elif calculation == "(chain, CG)":
+    percents.write("chain" + "," +  "change in CG´s in percent" + '\n')
+
+for key in changes:
+    percents.write(key + ',' + str(changes[key]) + '\n')
+
+percents.close()
```

Comment:
Common way to open files is a with-open statement: `with open (...):` as this prevents you from forgetting to close the file handler.